doc/classes/NavigationAgent2D.xml

```diff
@@ -198,6 +198,12 @@
 			The radius of the avoidance agent. This is the "body" of the avoidance agent and not the avoidance maneuver starting radius (which is controlled by [member neighbor_distance]).
 			Does not affect normal pathfinding. To change an actor's pathfinding radius bake [NavigationMesh] resources with a different [member NavigationMesh.agent_radius] property and use different navigation maps for each actor size.
 		</member>
+		<member name="simplify_epsilon" type="float" setter="set_simplify_epsilon" getter="get_simplify_epsilon" default="0.0">
+			The path simplification amount in worlds units.
+		</member>
+		<member name="simplify_path" type="bool" setter="set_simplify_path" getter="get_simplify_path" default="false">
+			If [code]true[/code] a simplified version of the path will be returned with less critical path points remove. The simplification amount is controlled by [member simplify_epsilon].
```

Comment:
```suggestion
			If [code]true[/code] a simplified version of the path will be returned with less critical path points removed. The simplification amount is controlled by [member simplify_epsilon].
```

After some more thoughts, I'd love if we could make this more concrete. What makes a path point critical, and what does "simplification amount" mean? What exactly does it mean when I set it to e.g. `2`?

Also, is there any better default than 0? (Does `0` even have an effect? If not, why is there a separate bool to turn the feature on?)

Reply:
The epsilon as mentioned is in world units so if a point is within this distance from the current simplified point segment it gets removed. If it is over this distance and also the point furthest away it becomes the new anchor point for the recursive segment simplification.

An epsilon of 0.0 is valid because it will still remove (most) path points that are on a perfect straight line. Though, It will not catch all of them when the line goes diagonal due to precision.

The main thing is, treating zero as "disabled" does not work because it is a valid epsilon. That is why there is a dedicated bool to enable the simplification.

Reply:
Thanks for the in-depth explanation! That mostly matches what I assumed.

> if a point is within this distance from the current simplified point segment it gets removed. If it is over this distance and also the point furthest away it becomes the new anchor point for the recursive segment simplification. That is what makes it a "critical" point.

Would be great if we could get some of this distilled into these docs. :)

> An epsilon of 0.0 […] will remove (most) path points that are on a perfect straight line

Maybe this too